engine/src/test/java/org/camunda/bpm/engine/test/bpmn/gateway/ParallelGatewayTest.java

```diff
@@ -373,4 +375,41 @@ public void testExplicitParallelGatewayAfterSignalBehavior() {
     // then
     assertEquals(3, taskService.createTaskQuery().count());
   }
+
+  @Deployment
+  @Test
+  @RequiredHistoryLevel(ProcessEngineConfiguration.HISTORY_FULL)
+  public void testParallelGatewayCancellationHistoryEvent() {
+    // given
+    // a process with one splitting and one merging parallel gateway and two parallel sequence flows between them
+    // one sequence flow has a wait state "Event_Wait", the other has none
```

Comment:
🔧 The new bpmn file can be replaced with the fluent builder (reducing introduced lines of code and improving readability(?)):
```suggestion
  @Test
  @RequiredHistoryLevel(ProcessEngineConfiguration.HISTORY_FULL)
  public void testParallelGatewayCancellationHistoryEvent() {
    // given
    // a process with one splitting and one merging parallel gateway and two parallel sequence flows between them
    // one sequence flow has a wait state "Event_Wait", the other has none
    testRule.deploy(Bpmn.createExecutableProcess("parallelProcess")
        .startEvent()
        .parallelGateway("Gateway_in")
          .intermediateCatchEvent("Event_Wait")
          .message("Message_1")
        .parallelGateway("Gateway_out")
        .endEvent()
        .moveToNode("Gateway_in")
        .connectTo("Gateway_out")
        .done());
```